joint_state_publisher/joint_state_publisher/joint_state_publisher.py

```diff
@@ -73,9 +114,11 @@ def init_collada(self, robot):
                     continue
 
                 if joint:
-                    limit = joint.getElementsByTagName('limits')[0]
-                    minval = float(limit.getElementsByTagName('min')[0].childNodes[0].nodeValue)
-                    maxval = float(limit.getElementsByTagName('max')[0].childNodes[0].nodeValue)
+                    limit_list = joint.getElementsByTagName('limits')
+                    if limit_list:
+                        limit = limit_list[0]
+                        minval = float(limit.getElementsByTagName('min')[0].childNodes[0].nodeValue)
+                        maxval = float(limit.getElementsByTagName('max')[0].childNodes[0].nodeValue)
```

Comment:
I'm not convinced we should do this particular fix here.  That is, this is a good fix to have, and I think we should make the fix, but I also think that if we do this we should do it consistently for all of `init_urdf`, `init_collada`, and `init_sdf`.  So I'll suggest we just do this in a separate PR.

Reply:
https://github.com/ros/joint_state_publisher/pull/94